virtwho/virt/virt.py

```diff
@@ -301,23 +301,42 @@ def __init__(self, logger, config, source=None, dest=None,
         super(IntervalThread, self).__init__()
 
     def wait(self, wait_time):
-        '''
+        """
         Wait `wait_time` seconds, could be interrupted by setting _terminate_event or _internal_terminate_event.
-        '''
+        """
         for i in range(wait_time):
             if self.is_terminated():
                 break
             time.sleep(1)
 
     def is_terminated(self):
         """
-
         @return: Returns true if either the internal terminate event is set or
                  the terminate event given in the init is set
         """
         return self._internal_terminate_event.is_set() or \
             self.terminate_event.is_set()
 
+    def is_registered(self):
+        """
+        This function tries to guess if system is registered according last attempt of
+        sending data to candlepin server.
+        :return: Return true, when virt-who was able to send data to candlepin server.
+        """
+
+        try:
+            is_registered = self.dest.get(key='is_registered', default=True)
```

Comment:
I didn't know that there could be more destination threads (for more candlepin servers?)

I will try to use different datastore.